pallets/parachain-staking/src/lib.rs

```diff
@@ -1546,13 +1559,23 @@ pub mod pallet {
 				if state.delegations.is_empty() {
 					// solo collator with no delegators
 					mint(amt_due, collator.clone());
+					extra_weight += T::OnCollatorPayout::on_collator_payout(
+						paid_for_round,
+						collator.clone(),
+						amt_due,
+					);
 				} else {
 					// pay collator first; commission + due_portion
 					let collator_pct = Perbill::from_rational(state.bond, state.total);
 					let commission = pct_due * collator_issuance;
 					amt_due = amt_due.saturating_sub(commission);
 					let collator_reward = (collator_pct * amt_due).saturating_add(commission);
 					mint(collator_reward, collator.clone());
+					extra_weight += T::OnCollatorPayout::on_collator_payout(
+						paid_for_round,
+						collator.clone(),
+						collator_reward,
+					);
```

Comment:
so we are minting the reward to the collator and then immediately after withdrawing it to the orbiters if they exist? I wonder if it is more efficient to check if the orbiters exist and transfer directly to them instead of first minting to the collator

Reply:
I think it's not a performance problem because the data are really written to the database only at the end of the block, here we move the rewards in the same block, only the events will remain.

Reply:
And I prefer not to modify the logic of the parachain-staking pallet for the moonbeam orbiters program, the idea is to add only handlers to "notify" the runtime